### DISCUSSION TITLE
Replace `maxDepth` with `defaultDepth` and/or allow depth overrides and/or something else?

### DISCUSSION THREAD

**Discussion by ssyberg** [payloadcms/payload#2978] [Feature Requests & Ideas]
Currently `maxDepth` works as expected but it has a number of drawbacks and I think some pretty light code changes could give us more granular control over depth. I haven't fully thought this through but I went through and removed all of the `maxDepths` throughout our system and now admin is slooooooow. The initial reason I did this is because I want to be able to set depth _per API call_ regardless of what the maxDepth is (i.e. I should be able to override the max) -- which means if we're overriding max well then it's not a max at all but simply a default. I think the same should be true of relationship depths. 

The above would be a great start, but would also be great to be able to specify relationship depth per field in a a query, though I'll take what I can get ;)

**Comment by jmikrut** [comment]
Hey @ssyberg — few questions for you. 

The Admin UI is set up to use `depth=0` basically everywhere - - so changing `defaultDepth` or `maxDepth` theoretically should not have any repercussions. There may be one or two spots where this is not the case, but overall, I wonder if there's something else at play here?

We could theoretically add another query param that says something like `ignoreMaxDepth`. Would that solve your issue above? But, overall, I would like to get to the bottom of why removing `maxDepth` has had a negative influence over your admin UI responsiveness. Any views in specific slower than others? Or is it everything? Could you inspect your Network panel for individual requests to the Payload API that are now slower?

**Reply by ssyberg** [reply]
Well that's very interesting indeed, I'll have to dive a bit deeper on this!
